rofi-{emoji,rbw}: move to pkgs/by-name, update and remove some antipatterns

acid-bong opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

acid-bong commented:
One question: how can I move `rofi-emoji`, since it has weird versioning here?

acid-bong committed:
rofi-rbw: move to pkgs/by-name

bew commented on pkgs/by-name/ro/rofi-wayland-unwrapped/package.nix:
```diff
@@ -0,0 +1,35 @@
+      wayland
+      wayland-protocols
+    ];
+  }
```
Please keep the meta, to have the wayland-specific description & homepage

acid-bong replied:
Yeah, I just noticed :sweat_smile: thx for pointing that out

bew commented on pkgs/by-name/ro/rofi-rbw/package.nix:
```diff
@@ -1,10 +1,7 @@
   lib,
-  buildPythonApplication,
+  python3Packages,
   fetchFromGitHub,
-  configargparse,
-  setuptools,
-  poetry-core,
   rbw,
 
```
Hmm this is technically a small breaking change, as someone who overridden those params would have to change the way they override.
I'm not sure how okay it is to change this 🤔
Maybe just add a mention in the release notes?

acid-bong replied:
@equirosa @dit7ya what are yall's thoughts on that?

acid-bong replied:
Python maintainers say it's the only way to move python programs to `pkgs/by-name`

acid-bong replied:
Only a message in Matrix. Anyway, packages in by-name are only called with the top-level callPackage, hence different scope

bew replied:
Yeah it makes sense, can you add a note in the next release note for the potential code change for overloading that pkg?

bew reviewed (commented):
Thanks for this! Just a few comments ;)

acid-bong committed:
rofi-emoji: move to pkgs/by-name, use v4

acid-bong committed:
rofi-emoji: add update script

cole-h reviewed (approved):
rofi-emoji-wayland still works for me after this*. LGTM.

(* I manually specified the `-plugin-path` out of laziness)

Mange reviewed (approved):
I don't mind the changes to `rofi-emoji`, although I do question overriding apps with completely different apps in some contexts.

acid-bong commented on pkgs/by-name/ro/rofi-emoji-wayland/package.nix:
```diff
@@ -0,0 +1,24 @@
+
+(rofi-emoji.override {
+  rofi-unwrapped = rofi-wayland-unwrapped;
+  xclip = wl-clipboard;
+  xdotool = wtype;
```
Oh, *both* X11 and Wayland, I missed that. Ight, I'll keep the original boolean toggles then.

Btw, why not use v4 source for both? Or did you remove X11 support from there?

acid-bong replied:
Yup, just checked out v4 on X11, works flawlessly

acid-bong commented on pkgs/by-name/ro/rofi-emoji/package.nix:
```diff
@@ -0,0 +1,94 @@
+  nix-update-script,
+}:
+let
+  rofi = if waylandSupport then rofi-wayland-unwrapped else rofi-unwrapped;
```
I'm using rofi-wayland by default, cuz it still works under X11 (in case someone wants to enable both X and WL tools)

Mange reviewed (approved):
Thank you for fixing this! ❤️

acid-bong commented:
Ight, now that both `rofi-wayland` and `rofi-wayland-unwrapped` are just one-line overrides, should I move them both to `top-level/all-packages`? or to `by-name`?

acid-bong committed:
rofi-rbw: modernize and cleanup

- pyprojectize the recipe
- buildInputs were useless
- wrapper paths and flags belonged in a `let..in`

acid-bong committed:
rofi-emoji: remove explicit patchShebang-ing of clipboard-adapter.sh

it's executable, so it gets patchShebang-ed in fixupPhase anyway

acid-bong committed:
rofi-emoji: 4.0.0 -> 4.1.0

acid-bong commented:
> No conflicts with base branch

fucking finally :weary::pray:

cole-h reviewed (approved):
Seems fine to me. Let's get this in before any more painful merge conflicts ;)
